storm-shim-10x/src/main/java/storm/mesos/shims/LocalStateShim.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package storm.mesos.shims;
+
+import backtype.storm.generated.ComponentObject;
+import backtype.storm.generated.JavaObject;
+import backtype.storm.generated.JavaObjectArg;
+import backtype.storm.utils.LocalState;
+import org.apache.thrift7.TBase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LocalStateShim implements ILocalStateShim {
+  LocalState _state;
+
+  public LocalStateShim(String localDir) throws IOException {
+    _state = new LocalState(localDir);
+  }
+
+  public String get(String key) {
+    TBase tBase = _state.get(key);
+    if (tBase != null) {
+      JavaObject jo = ((JavaObject) tBase);
```

Comment:
Do you have any references to the Mesos State APIs' docs?  I've found some references to some frameworks that supposedly use them (e.g., in Aurora), but I've had a really hard time finding some simple API.  i.e., I expect a trivial API with add/get/delete into the Mesos State managed by the mesos-master.  But I think I might misunderstand what they are providing with the "State API" and that it is simply providing an interface layer on top of a ton of junk you have to set up yourself.